torch_xla/core/xla_model.py

```diff
@@ -427,15 +427,29 @@ def all_gather(value, dim=0):
     A tensor which has, in the ``dim`` dimension, all the values from the
     participating replicas.
   """
+
+  def get_pad_indices(groups):
```

Comment:
I am not sure we need a single use nested function there.
Also, I am not in love with using index+exception.
This is less code and flows better IMHO:

```Python
ordinal = get_ordinal()                                                                                                                                                                                                                     
if groups is None:                                                                                                                                                                                                                          
  left, right = ordinal, xrt_world_size() - 1 - ordinal                                                                                                                                                                                     
else:                                                                                                                                                                                                                                       
  ordinals = dict()                                                                                                                                                                                                                         
  for g in groups:                                                                                                                                                                                                                          
    for i, x in enumerate(g):                                                                                                                                                                                                               
      ordinals[x] = (i, len(g) - 1 - i)                                                                                                                                                                                                     
  left, right = ordinals[ordinal] 
```

Reply:
thanks, applied this and tested.